### PR TITLE
alpine/musl fixes

### PR DESCRIPTION
**Pull request by rotschopf**
I stumbled over two failures while building the streamer in an alpine docker image.
Here are the fixes i came up with.
Still builds on my laptop with kubuntu 16.04.

### CODE REVIEW

File: mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c [hunk 1]
```diff
@@ -234,7 +234,7 @@ int input_init(input_parameter *param, int id)
         case 2:
         case 3:
             DBG("case 2,3\n");
-            dev = canonicalize_file_name(optarg);
+            dev = realpath(optarg, NULL);
```
Comment (virtuald) [inline mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c]: This seems wrong, isn't that a function defined in mjpg-streamer somewhere? (No computer atm)

If it's not, then should use an ifdef to detect that it doesn't exist.

Comment (jacksonliam) [inline mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c]: I thought it was part of stdlib? Not sure why alpine doesn't have it? 

Man says the two calls should be equivalent though http://man7.org/linux/man-pages/man3/canonicalize_file_name.3.html

Comment (rotschopf) [inline mjpg-streamer-experimental/plugins/input_uvc/input_uvc.c]: Well its a GNU extension. Maybe musl (the clib of alpine) does not come with those!?
At least a search does come up with no results:
https://github.com/bminor/musl/search?utf8=%E2%9C%93&q=canonicalize_file_name

realpath on the other hand is POSIX and as you already pointed out the man page says those calls are equivalent. So I don't see a reason not to switch to realpath.

**Comment by jacksonliam** [comment]
I've tested it and the realpath thing seems to work fine, even with symlinks or relative paths. I can't see the extra include doing any harm - thanks for the PR!

**Pull request merged**